amqp-impl/src/main/java/io/streamnative/pulsar/handlers/amqp/QueueServiceImpl.java

```diff
@@ -0,0 +1,293 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.streamnative.pulsar.handlers.amqp;
+
+import static org.apache.qpid.server.protocol.ErrorCodes.INTERNAL_ERROR;
+
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.qpid.server.protocol.ErrorCodes;
+import org.apache.qpid.server.protocol.v0_8.AMQShortString;
+import org.apache.qpid.server.protocol.v0_8.FieldTable;
+import org.apache.qpid.server.protocol.v0_8.transport.AMQMethodBody;
+import org.apache.qpid.server.protocol.v0_8.transport.MethodRegistry;
+import org.apache.qpid.server.protocol.v0_8.transport.QueueDeclareOkBody;
+import org.apache.qpid.server.protocol.v0_8.transport.QueueDeleteOkBody;
+
+/**
+ * Logic of queue.
+ */
+@Slf4j
+public class QueueServiceImpl implements QueueService {
+    private ExchangeContainer exchangeContainer;
+    private QueueContainer queueContainer;
+
+    public QueueServiceImpl(ExchangeContainer exchangeContainer,
+                            QueueContainer queueContainer) {
+        this.exchangeContainer = exchangeContainer;
+        this.queueContainer = queueContainer;
+    }
+
+    @Override
+    public void queueDeclare(AmqpChannel channel, AMQShortString queue, boolean passive, boolean durable,
+                             boolean exclusive, boolean autoDelete, boolean nowait, FieldTable arguments) {
+        int channelId = channel.getChannelId();
+        AmqpConnection connection = channel.getConnection();
+        if (log.isDebugEnabled()) {
+            log.debug("RECV[{}] QueueDeclare[ queue: {}, passive: {}, durable:{}, "
+                            + "exclusive:{}, autoDelete:{}, nowait:{}, arguments:{} ]",
+                    channelId, queue, passive, durable, exclusive, autoDelete, nowait, arguments);
+        }
+        if ((queue == null) || (queue.length() == 0)) {
+            queue = AMQShortString.createAMQShortString("tmp_" + UUID.randomUUID());
+        }
+        AMQShortString finalQueue = queue;
+        CompletableFuture<AmqpQueue> amqpQueueCompletableFuture =
+                queueContainer.asyncGetQueue(connection.getNamespaceName(), finalQueue.toString(), !passive);
+        amqpQueueCompletableFuture.whenComplete((amqpQueue, throwable) -> {
+            if (throwable != null) {
+                log.error("Get Topic error:{}", throwable.getMessage());
```

Comment:
Print all stracktrace and give a meaningful error message.

Reply:
Done.